### DISCUSSION TITLE
Outline Menu Commands to Move Lines Up or Down

### DISCUSSION THREAD

**Discussion by axodys** [vincode-io/Zavala#87] [Ideas]
Being able to reorganize lines up or down within their hierarchy via menu command/keystroke is a pretty standard outliner feature. Dave Winer's Little Outliner uses Cmd-U to Move Up the active line and Cmd-D to Move Down, but alternatively Omni Outliner uses Ctrl-Cmd-⬆️ and Ctrl-Cmd-⬇️. I see you can perform this action by clicking and dragging in Zavala, but on a Mac this is something you want to do via a key combo just like indenting.

**Comment by vincode-io** [comment]
I'd seen that feature in Omni Outliner, but never used it much myself. I wasn't aware that other outliners had it as well. I'm taking a cautious approach to adding features. It is easier to give more than to take away.

The bad news is that it is too late to get this into the 1.0 release. The good news is that I plan to get 1.1 out fairly quickly, within a month or two, and I'l make sure this feature is in it.

**Reply by philiprenich** [reply]
Another option for the keybinds are `opt + ↑` and `opt + ↓`. This is how VS Code does it. Different software paradigm and I'm not sure how universal that is either, so it may be better to match another outliner instead of a coding IDE. But that's what I'd instinctively try first. 🤷🏽

**Comment by latvia234** [comment]
I use this a lot with Dynalist where the keyboard shortcuts are cmd+↑ and cmd+↓

I just remap the keyboard shortcuts if needed.

**Comment by Olligarch** [comment]
I also use this function and shortcut all the time with Cloud Outliner and iThoughtsX. It would be really great if this could be included in Zavala.

**Comment by vincode-io** [comment]
This feature was added in Zavala 1.1. Right now we are testing it on iOS. You should have access to this command on iOS using the hardware keyboard. Alternatively you can use the new toolbar buttons to perform the same actions.

It was also added to the macOS version as menu items, toolbar items, and keyboard shortcuts. Unfortunately because we don't have TestFlight for the Mac yet, we can't test there. Once macOS 12 gets released, the situation will be different there.

**Reply by Olligarch** [reply]
Perfect! I can't wait for version 1.1! Great work!
